drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -616,10 +616,27 @@ static int airoc_mgmt_ap_enable(const struct device *dev, struct wifi_connect_re
 			params->channel);
 	}
 
-	if (params->psk_length == 0) {
+	switch (params->security) {
+	case WIFI_SECURITY_TYPE_NONE:
 		security = WHD_SECURITY_OPEN;
-	} else {
+		break;
+	case WIFI_SECURITY_TYPE_WEP:
```

Comment:
I am confused by the commit log, the existing code defaulted to WPA2-PSK is a passphrase is given, so, new code should check the user requested algo and reject with appropriate error, but I only see mapping here, am I missing something?

Reply:
Sorry. I don't get your point well.
When I configured WPA3-SAE, original code configures WPA2-PSK since it check if pks_length is 0 or not.
Then, I modified to check security type in order to use WPA3-SAE. Am I missing anything.

Reply:
Sure, I understand that, but I was reading the commit log and it was matching older implementation. IIUC, the limitation in the chip is now gone and it supports all security algos? If yes, the fix is good, commit log can be re-worded.